docs/new_agents.rst

```diff
@@ -1,3 +1,390 @@
+As an example, we use the registry and aggregator agents, which are
+necessary for taking any data with OCS, as well as the Arduino agent.
+
+::
+
+  hosts:
+
+    grumpy: {
+
+        'agent-instances': [
+            # Core OCS Agents
+            {'agent-class': 'RegistryAgent',
+             'instance-id': 'registry',
+             'arguments': []},
+            {'agent-class': 'AggregatorAgent',
+             'instance-id': 'aggregator',
+             'arguments': [['--initial-state', 'record'],
+                           ['--time-per-file', '3600'],
+                           ['--data-dir', '/data/']]},
+
+            # Arduino
+            {'agent-class': 'ArduinoAgent',
+             'instance-id': 'arduino',
+             'arguments': []},
+        ]
+    }
+
+When adding a new Agent, the ``'agent-class'`` entry should match the name of
+your class in the Agent file. The ``'arguments'`` entry should match any
+arguments that you added to ``pgroup`` at the end of your Agent file.
+
+In this example, the ``'agent-instances'`` are found under a host called 
+``grumpy``, which in this case is the name of the host computer. However, when 
+writing an Agent that will be broadly useful, we may choose to Dockerize the 
+Agent (and its dependencies). For more on this, see the Docker section of this 
+documentation.
+
+
+Docker
+------
+A Docker container creates a virtual environment in which you can package 
+applications with their libraries and dependencies. OCS is sometimes installed 
+in a Docker container (for ease of installation). For Agents that are not 
+meant solely to be used with one lab computer, it can be useful to add them to a 
+Docker container as well. This requires creating a ``Dockerfile`` for your Agent 
+and adding the Agent capabilites to your OCS Docker container in a 
+``docker-compose.yml`` file. Adding your Agent in the ``docker-compose.yml`` file 
+will also allow you to view your data feed when you run the Agent.
+
+To create a ``Dockerfile``, change directories to the directory containing your 
+Agent file. Within this directory, create a file called ``Dockerfile``. The format 
+of this file is as follows (using the Arduino as an example):
+
+::
+
+        # SOCS Arduino Agent
+        # socs Agent container for interacting with an Arduino
+
+        # Use socs base image
+        FROM socs:latest
+
+        # Set the working directory to registry directory
+        WORKDIR /app/agents/arduino/
+
+        # Copy this agent into the app/agents directory
+        COPY . /app/agents/arduino/
+
+        # Run registry on container startup
+        ENTRYPOINT ["python3", "-u", "arduino_agent.py"]
+
+
+In this case, the ``WORKDIR``, ``COPY``, and ``ENTRYPOINT`` arguments are all set 
+specifically to the correct directories and files for the Arduino agent. You can 
+additionally connect the container to a Crossbar (WAMP) server; see the Sisock 
```

Comment:
You might be referring to these lines found in some other agent Dockerfiles:
```
CMD ["--site-hub=ws://sisock-crossbar:8001/ws", \
     "--site-http=http://sisock-crossbar:8001/call"]
```
This is providing a "sensible default" for the crossbar connection, else it'll use whatever is configured in your ocs-site-configs file. This maybe should be included in new Dockerfiles. It'll almost guarantee the system works out of the box for a single node system, but might need overriding in a multi-node system.

Either way, connection the crossbar server is *required* for the Agent to function, so this should be rephrased.